From fdfbb08a32525a455f632e2dc5cb2ecc7be42528 Mon Sep 17 00:00:00 2001
From: Eric Roca <eric.roca@gmail.com>
Date: Fri, 12 Nov 2021 22:30:36 +0100
Subject: [PATCH] Changes the scrollback keybinding.

---
 config.def.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/config.def.h b/config.def.h
index 536db70..641d2bd 100644
--- a/config.def.h
+++ b/config.def.h
@@ -11,6 +11,6 @@ struct rule rules[] = {
 	{"\033[5;2~",   SCROLL_UP,   -1},	/* [Shift] + [PageUP] */
 	{"\033[6;2~",   SCROLL_DOWN, -1},	/* [Shift] + [PageDown] */
 	/* mouse binding shadows ^E and ^Y, so it's disabled by default */
-	//{"\031",        SCROLL_UP,    1},	/* mouse wheel up */
-	//{"\005",        SCROLL_DOWN,  1},	/* mouse wheel Down */
+	{"\031",        SCROLL_UP,    1},	/* mouse wheel up */
+	{"\033[1;6B",   SCROLL_DOWN,  1},	/* mouse wheel Down */
 };
--
2.46.2

