From 5797061eb1df8730eac9d88853f98051d6e2c68a Mon Sep 17 00:00:00 2001
From: Eric Roca <eric.roca@gmail.com>
Date: Mon, 30 Sep 2024 19:34:05 +0200
Subject: [PATCH 7/7] Adds personal configuration

---
 config.def.h | 95 ++++++++++++++++++++++++++--------------------------
 1 file changed, 48 insertions(+), 47 deletions(-)

diff --git a/config.def.h b/config.def.h
index 1ee011a..b745774 100644
--- a/config.def.h
+++ b/config.def.h
@@ -5,7 +5,7 @@
  *
  * font: see http://freedesktop.org/software/fontconfig/fontconfig-user.html
  */
-static char *font = "Liberation Mono:pixelsize=12:antialias=true:autohint=true";
+static char *font = "Fira Code Nerd Font Mono:pixelsize=14:antialias=true:autohint=true";
 /* Spare fonts */
 static char *font2[] = {
 /*	"Inconsolata for Powerline:pixelsize=12:antialias=true:autohint=true", */
@@ -24,7 +24,7 @@ static int borderpx = 2;
 static char *shell = "/bin/sh";
 char *utmp = NULL;
 /* scroll program: to enable use a string like "scroll" */
-char *scroll = NULL;
+char *scroll = "/usr/bin/scroll";
 char *stty_args = "stty raw pass8 nl -echo -iexten -cstopb 38400";

 /* identification sequence returned in DA and DECID */
@@ -102,45 +102,40 @@ unsigned int tabspaces = 8;
 float alpha = 0.8;

 /* Terminal colors (16 first used in escape sequence) */
-static const char *colorname[] = {
-	/* 8 normal colors */
-	"black",
-	"red3",
-	"green3",
-	"yellow3",
-	"blue2",
-	"magenta3",
-	"cyan3",
-	"gray90",
-
-	/* 8 bright colors */
-	"gray50",
-	"red",
-	"green",
-	"yellow",
-	"#5c5cff",
-	"magenta",
-	"cyan",
-	"white",
-
-	[255] = 0,
-
-	/* more colors can be added after 255 to use with DefaultXX */
-	"#cccccc",
-	"#555555",
-	"gray90", /* default foreground colour */
-	"black", /* default background colour */
-};
-
-
-/*
- * Default colors (colorname index)
- * foreground, background, cursor, reverse cursor
- */
-unsigned int defaultfg = 258;
-unsigned int defaultbg = 259;
-unsigned int defaultcs = 256;
-static unsigned int defaultrcs = 257;
+const char *colorname[] = {
+
+  /* 8 normal colors */
+  [0] = "#0f1118", /* black   */
+  [1] = "#31526E", /* red     */
+  [2] = "#4E5A63", /* green   */
+  [3] = "#A09B67", /* yellow  */
+  [4] = "#366B8D", /* blue    */
+  [5] = "#4A748D", /* magenta */
+  [6] = "#5A92A3", /* cyan    */
+  [7] = "#9dcadb", /* white   */
+
+  /* 8 bright colors */
+  [8]  = "#6d8d99",  /* black   */
+  [9]  = "#31526E",  /* red     */
+  [10] = "#4E5A63", /* green   */
+  [11] = "#A09B67", /* yellow  */
+  [12] = "#366B8D", /* blue    */
+  [13] = "#4A748D", /* magenta */
+  [14] = "#5A92A3", /* cyan    */
+  [15] = "#9dcadb", /* white   */
+
+  /* special colors */
+  [256] = "#0f1118", /* background */
+  [257] = "#9dcadb", /* foreground */
+  [258] = "#9dcadb",     /* cursor */
+ };
+
+/* Default colors (colorname index)
+ * foreground, background, cursor */
+unsigned int defaultbg = 0;
+unsigned int defaultfg = 257;
+unsigned int defaultcs = 258;
+static unsigned int defaultrcs= 258;

 /*
  * Default shape of cursor
@@ -178,17 +173,21 @@ static unsigned int defaultattr = 11;
  */
 static uint forcemousemod = ShiftMask;

+/*
+ * User-defined functions
+*/
+static char *openurlcmd[] = { "/bin/sh", "-c", "st-urlhandler -o", "externalpipe", NULL };
+static char *copyurlcmd[] = { "/bin/sh", "-c", "st-urlhandler -c", "externalpipe", NULL };
+
 /*
  * Internal mouse shortcuts.
  * Beware that overloading Button1 will disable the selection.
  */
 static MouseShortcut mshortcuts[] = {
-	/* mask                 button   function        argument       release */
-	{ XK_ANY_MOD,           Button2, selpaste,       {.i = 0},      1 },
-	{ ShiftMask,            Button4, ttysend,        {.s = "\033[5;2~"} },
-	{ XK_ANY_MOD,           Button4, ttysend,        {.s = "\031"} },
-	{ ShiftMask,            Button5, ttysend,        {.s = "\033[6;2~"} },
-	{ XK_ANY_MOD,           Button5, ttysend,        {.s = "\005"} },
+	/* mask                 button   function        argument         release */
+	{ XK_ANY_MOD,           Button2, selpaste,       {.i = 0},        1 },
+	{ XK_ANY_MOD,           Button4, ttysend,        {.s = "\031"}      },
+	{ XK_ANY_MOD,           Button5, ttysend,        {.s = "\033[1;6B"} },
 };

 /* Internal keyboard shortcuts. */
@@ -209,6 +208,8 @@ static Shortcut shortcuts[] = {
 	{ TERMMOD,              XK_Y,           selpaste,       {.i =  0} },
 	{ ShiftMask,            XK_Insert,      selpaste,       {.i =  0} },
 	{ TERMMOD,              XK_Num_Lock,    numlock,        {.i =  0} },
+    { TERMMOD,              XK_U,           externalpipe,   {.v =  openurlcmd } },
+    { TERMMOD,              XK_O,           externalpipe,   {.v =  copyurlcmd } },
 };

 /*
--
2.46.2

